extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/AvroFeatureDataAdapter.java

```diff
@@ -0,0 +1,123 @@
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>());
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final VisibilityManagement<SimpleFeature> visibilityManagement ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>(),
+				null,
+				visibilityManagement);
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final List<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>> customIndexHandlers ) {
+		super(
+				type,
+				customIndexHandlers,
+				null,
+				new JsonDefinitionColumnVisibilityManagement<SimpleFeature>());
+	}
+
+	public AvroFeatureDataAdapter(
+			final SimpleFeatureType type,
+			final FieldVisibilityHandler<SimpleFeature, Object> fieldVisiblityHandler ) {
+		super(
+				type,
+				new ArrayList<PersistentIndexFieldHandler<SimpleFeature, ? extends CommonIndexValue, Object>>(),
+				fieldVisiblityHandler,
+				new JsonDefinitionColumnVisibilityManagement<SimpleFeature>());
+	}
+
+	@Override
+	protected List<NativeFieldHandler<SimpleFeature, Object>> typeToFieldHandlers(
+			final SimpleFeatureType type ) {
+		final List<NativeFieldHandler<SimpleFeature, Object>> nativeHandlers = new ArrayList<NativeFieldHandler<SimpleFeature, Object>>(
+				1);
+
+		nativeHandlers.add(new AvroFeatureAttributeHandler());
+		return nativeHandlers;
+	}
+
+	@Override
+	protected IndexFieldHandler<SimpleFeature, Time, Object> getTimeRangeHandler(
+			final SimpleFeatureType featureType ) {
+		final TimeDescriptors timeDescriptors = inferTimeAttributeDescriptor(featureType);
+
+		if ((timeDescriptors.getStartRange() != null) && (timeDescriptors.getEndRange() != null)) {
+			return (new FeatureTimeRangeHandler(
+					new FeatureAttributeHandler(
+							timeDescriptors.getStartRange()),
+					new FeatureAttributeHandler(
+							timeDescriptors.getEndRange()),
+					new AvroAdaptorProxyFieldLevelVisibilityHandler(
+							timeDescriptors.getStartRange().getLocalName(),
+							this)));
+		}
+		else if (timeDescriptors.getTime() != null) {
+			// if we didn't succeed in identifying a start and end time,
+			// just grab the first attribute and use it as a timestamp
+			return new FeatureTimestampHandler(
+					timeDescriptors.getTime(),
+					new AvroAdaptorProxyFieldLevelVisibilityHandler(
+							timeDescriptors.getTime().getLocalName(),
+							this));
+		}
+		return null;
+	}
+
+	@Override
+	public FieldReader<Object> getReader(
+			final ByteArrayId fieldId ) {
+		return (FieldReader<Object>) new AvroFeatureReader();
+	}
+
+	@Override
+	public FieldWriter<SimpleFeature, Object> getWriter(
+			final ByteArrayId fieldId ) {
+		return (FieldWriter<SimpleFeature, Object>) new AvroFeatureWriter();
+	}
+
+	@Override
+	protected RowBuilder<SimpleFeature, Object> newBuilder() {
+		return new AvroAttributeRowBuilder(
+				AvroFeatureAttributeHandler.FIELD_ID);
```

Comment:
If you make the suggested changes to the AvroAttributeRowBuilder, you won't need to pass the field id here.